lib/questions.rb

```diff
@@ -24,31 +31,37 @@ def reverse_every_element_in_array(array)
 # [['Bob', 'Clive'], ['Bob', 'Dave'], ['Clive', 'Dave']]
 # make sure you don't have the same pairing twice, 
 def every_possible_pairing_of_students(array)
+  array.combination(2).to_a
 end
 
 # discard the first 3 elements of an array, 
 # e.g. [1, 2, 3, 4, 5, 6] becomes [4, 5, 6]
 def all_elements_except_first_3(array)
+  array.drop(3)
 end
 
 # add an element to the beginning of an array
 def add_element_to_beginning_of_array(array, element)
+  array.unshift(element)
 end
 
 # sort an array of words by their last letter, e.g.
 # ['sky', 'puma', 'maker'] becomes ['puma', 'maker', 'sky']
 def array_sort_by_last_letter_of_word(array)
+  array.sort! {|a,b| a.byteslice(a.length-1) <=> b.byteslice(b.length-1)}
```

Comment:
Space missing after comma.<br>Surrounding space missing for operator '-'.<br>Space between { and | missing.<br>Space missing inside }.